sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/WriteToDirectory.scala

```diff
@@ -0,0 +1,177 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.execution
+
+import java.io.IOException
+
+import org.apache.hadoop.fs.{FileSystem, Path}
+import org.apache.hadoop.hive.common.FileUtils
+import org.apache.hadoop.hive.conf.HiveConf.ConfVars
+import org.apache.hadoop.hive.ql.Context
+import org.apache.hadoop.hive.ql.plan.TableDesc
+import org.apache.hadoop.hive.serde2.Serializer
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspectorUtils.ObjectInspectorCopyOption
+import org.apache.hadoop.hive.serde2.objectinspector.{ObjectInspectorUtils, StructObjectInspector}
+import org.apache.hadoop.mapred.{FileOutputCommitter, FileOutputFormat, JobConf}
+import org.apache.spark.annotation.DeveloperApi
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.catalyst.expressions._
+import org.apache.spark.sql.execution.{SparkPlan, UnaryNode}
+import org.apache.spark.sql.hive.HiveShim._
+import org.apache.spark.sql.hive.{HiveContext, HiveInspectors, HiveShim, SparkHiveWriterContainer, ShimFileSinkDesc => FileSinkDesc}
+import org.apache.spark.{SerializableWritable, SparkContext, TaskContext}
+
+import scala.collection.JavaConversions._
+
+/**
+ * :: DeveloperApi ::
+ */
+@DeveloperApi
+case class WriteToDirectory(
+   path: String,
+   child: SparkPlan,
+   isLocal: Boolean,
+   desc: TableDesc) extends UnaryNode with HiveInspectors {
+
+  @transient val hiveContext = sqlContext.asInstanceOf[HiveContext]
+  @transient private lazy val context = new Context(hiveContext.hiveconf)
+  @transient lazy val outputClass = newSerializer(desc).getSerializedClass
+
+  def output = child.output
+  
+  private def newSerializer(tableDesc: TableDesc): Serializer = {
+    val serializer = tableDesc.getDeserializerClass.newInstance().asInstanceOf[Serializer]
+    serializer.initialize(null, tableDesc.getProperties)
+    serializer
+  }
+
+  // maybe we can make it as a common method, share with `InsertIntoHiveTable`
```

Comment:
Yes in InsertIntoHiveTable we can write data using arbitrary SerDes to a file system. But we can not reuse InsertIntoHiveTable to share the same physical plan since 
1 InsertIntoHiveTable and  WriteToDirectory have different input
2 InsertIntoHiveTable has complex logical to handle dynamic partitioning

But i think we can extract a common interface(SaveAsHiveFile) to reuse the code of writing data to file system.